assembly/std/arguments.ts

```diff
@@ -0,0 +1,165 @@
+import {Address} from './address';
+import {ByteArray} from '@massalabs/as/assembly/byteArray';
+
+/**
+ * Args for remote function call
+ */
+export class Args {
+  private offset: i64 = 0;
+  private argsString: string = '';
+
+  /**
+   *
+   * @param {string} argsString
+   * @param {Array<string>} types
+   */
+  constructor(argsString: string = '') {
+    this.argsString = argsString;
+  }
+
+  /**
+   *
+   * @return {string} the serialized string
+   */
+  serialize(): string {
+    return this.argsString;
+  }
+
+  // getters
+
+  /**
+   *
+   * @return {Address} the address
+   */
+  nextAddress(): Address {
+    const address = new Address();
+    this.offset = address.fromStringSegment(
+      this.argsString,
+      this.offset as i32,
+    );
+    return address;
+  }
+
+  /**
+   *
+   * @return {string} the string
+   */
+  nextString(): string {
+    let offset: i32 = this.offset as i32;
+    const length = u8(this.argsString.codePointAt(offset));
+    const result = this.argsString.slice(offset + 1, offset + length + 1);
```

Comment:
Why the last +1 ?
Using a variable such as `end` will make it clearer that the second index is not the last element, but the first after (hence the classical end denomination) and it will avoid recopying 'offset + length + 1' twice :-)

Reply:
okay I'll try something